agent/engine/task_manager.go

```diff
@@ -624,45 +674,61 @@ func (mtask *managedTask) handleContainerStoppedTransitionError(event DockerCont
 	return true
 }
 
-// progressContainers tries to step forwards all containers that are able to be
+// progressTask tries to step forwards all containers and resources that are able to be
 // transitioned in the task's current state.
 // It will continue listening to events from all channels while it does so, but
 // none of those changes will be acted upon until this set of requests to
 // docker completes.
 // Container changes may also prompt the task status to change as well.
-func (mtask *managedTask) progressContainers() {
-	seelog.Debugf("Managed task [%s]: progressing containers in task", mtask.Arn)
+func (mtask *managedTask) progressTask() {
+	seelog.Debugf("Managed task [%s]: progressing containers and resources in task", mtask.Arn)
 	// max number of transitions length to ensure writes will never block on
 	// these and if we exit early transitions can exit the goroutine and it'll
 	// get GC'd eventually
-	transitionChange := make(chan struct{}, len(mtask.Containers))
-	transitionChangeContainer := make(chan string, len(mtask.Containers))
+	transitionChange := make(chan struct{}, len(mtask.Containers)+len(mtask.Resources))
+	transitionChangeEntity := make(chan string, len(mtask.Containers)+len(mtask.Resources))
 
-	anyCanTransition, transitions, reasons := mtask.startContainerTransitions(
+	anyResourceTransition, resTransitions := mtask.startResourceTransitions(
+		func(resource taskresource.TaskResource, nextStatus taskresource.ResourceStatus) {
+			mtask.engine.transitionResource(mtask.Task, resource, nextStatus)
+			transitionChange <- struct{}{}
+			transitionChangeEntity <- resource.GetName()
+		})
+
+	anyContainerTransition, contTransitions, contReasons := mtask.startContainerTransitions(
 		func(container *api.Container, nextStatus api.ContainerStatus) {
 			mtask.engine.transitionContainer(mtask.Task, container, nextStatus)
 			transitionChange <- struct{}{}
-			transitionChangeContainer <- container.Name
+			transitionChangeEntity <- container.Name
 		})
 
-	if !anyCanTransition {
-		if !mtask.waitForExecutionCredentialsFromACS(reasons) {
+	if !anyContainerTransition {
```

Comment:
I think you are missing the case where all containers are waiting for the resource. You can either check the `anyResourceTransition` or add a function like `waitForExecutionCredentialsFromACS` to check whether the containers are waiting for resource.